browse/services/util/formats.py

```diff
@@ -0,0 +1,161 @@
+"""Functions that support determintation of dissemination formats."""
+import os
+import re
+from operator import itemgetter
+from typing import Dict, List, Optional
+
+import tarfile
+from tarfile import ReadError, CompressionError
+
+import logging
+logger = logging.getLogger(__name__)
+
+# List of tuples containing the valid source file name extensions and their
+# corresponding dissemintation formats.
+# There are minor performance implications in the ordering when doing
+# filesystem lookups, so the ordering here should be preserved.
+VALID_SOURCE_EXTENSIONS = [
```

Comment:
Also could be shared with `arxiv-compiler` so lift to `arxiv.base`?

Reply:
Sounds good.